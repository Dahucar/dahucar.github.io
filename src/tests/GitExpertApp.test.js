import React from 'react';
import { shallow } from 'enzyme';
import GitExpertApp from '../components/GifExpertApp';

describe('Testing on GitExpertApp Component', () => {
  
  test('Should generate snapshot of GitExpertApp component', () => {
    const wrapper = shallow( <GitExpertApp /> );  
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('Should be displeyed a list categories', () => {
    const categories = ['Halo', 'GOD', 'Nedd For Speed'];
    const wrapper = shallow( <GitExpertApp defaultCategories={ categories } /> );  
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
  });
});
