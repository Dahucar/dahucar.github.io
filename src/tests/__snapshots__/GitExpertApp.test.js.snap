// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on GitExpertApp Component Should be displeyed a list categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Halo"
      key="Halo"
    />
    <GifGrid
      category="GOD"
      key="GOD"
    />
    <GifGrid
      category="Nedd For Speed"
      key="Nedd For Speed"
    />
  </ol>
</Fragment>
`;

exports[`Testing on GitExpertApp Component Should generate snapshot of GitExpertApp component 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
